src/Bridge/Doctrine/Orm/Filter/SearchFilter.php

```diff
@@ -272,7 +272,8 @@ protected function filterProperty(string $property, $value, QueryBuilder $queryB
 
         if ($metadata->isCollectionValuedAssociation($association)) {
             $associationAlias = QueryBuilderHelper::addJoinOnce($queryBuilder, $queryNameGenerator, $alias, $association);
-            $associationField = 'id';
+            $targetEntityMetaData = $this->getClassMetadata($metadata->getAssociationMapping($association)['targetEntity']);
+            $associationField = $targetEntityMetaData->getIdentifier()[0];
```

Comment:
it'll not work with mutliple identifiers but I guess it's better then before haha.

I'm :+1: for the change